lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -127,81 +265,162 @@ def self.bump_version_release
+        #
+        # @return [String] The value of the versionName attribute as found in the build.gradle file and for this section.
+        #
+        def self.get_version_name_from_gradle_file(file_path, section)
+          res = get_keyword_from_gradle_file(file_path, section, "versionName")
+          res = res.tr('\"', '') unless res.nil?
           return res
         end
   
-        def self.get_version_build_from_file(file_path, section)
-          res = get_data_from_file(file_path, section, "versionCode")
-          res = res.split(' ')[1].to_i
-          return res
+        # Extract the versionCode rom a build.gradle file
+        #
+        # @param [String] file_path The path to the `.gradle` file
+        # @param [String] section The name of the section we expect the keyword to be in, e.g. "defaultConfig" or "vanilla"
+        #
+        # @return [String] The value of the versionCode attribute as found in the build.gradle file and for this section.
+        #
+        def self.get_version_build_from_gradle_file(file_path, section)
+          res = get_keyword_from_gradle_file(file_path, section, "versionCode")
+          return res.to_i
         end
 
-        # FIXME: This implementation is very fragile. This should be done parsing the file in a proper way. 
-        # Leveraging gradle itself is probably the easiest way.
-        def self.get_data_from_file(file_path, section, keyword)
+        # Extract the value for a specific keyword in a specific section of a `.gradle` file
+        #
+        # @todo: This implementation is very fragile. This should be done parsing the file in a proper way. 
+        #        Leveraging gradle itself is probably the easiest way.
+        #
+        # @param [String] file_path The path of the `.gradle` file to extract the value from
+        # @param [String] section The name of the section from which we want to extract this keyword from. For example `defaultConfig` or `myFlavor`
+        # @param [String] keyword The keyword (key name) we want the value for
+        #
+        # @return [String] Returns the value for that keyword in the section of the `.gradle` file, or nil if not found.
+        #
+        def self.get_keyword_from_gradle_file(file_path, section, keyword)
           found_section = false
           File.open(file_path, 'r') do |file|
             file.each_line do |line|
               if !found_section
-                if (line.include? section) 
+                if line.include?(section)
                   found_section = true
                 end
               else
-                if (line.include? keyword)
-                  return line unless line.include?("\"#{keyword}\"") or line.include?("P#{keyword}")
+                if line.include?(keyword) && !line.include?("\"#{keyword}\"") && !line.include?("P#{keyword}")
+                  return line.split(' ')[1]
```

Comment:
All call sites were doing the `split(' ')[1]`, so I figured I'd DRY it here – which actually makes more sense so that the method returns the keyword value, and not the "`.gradle` line containing the keyword, with potentially some indentation and spaces in front"